### PR TITLE
Spark 3.5: Remaining tests migrated to Junit5 

### PR DESCRIPTION
**Pull request by chinmay-bhat**
This PR migrates remaining tests in Spark3.5 to JUnit5. 

Issue: #9086

### CODE REVIEW

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/TestChangelogIterator.java [hunk 1]
```diff
@@ -196,10 +196,10 @@ public void testUpdatedRowsWithDuplication() {
...
         ChangelogIterator.computeUpdates(rowsWithDuplication.iterator(), SCHEMA, IDENTIFIER_FIELDS);
 
-    assertThrows(
-        "Cannot compute updates because there are multiple rows with the same identifier fields([id, name]). Please make sure the rows are unique.",
-        IllegalStateException.class,
-        () -> Lists.newArrayList(iterator));
+    assertThatThrownBy(() -> Lists.newArrayList(iterator))
+        .as(
+            "Cannot compute updates because there are multiple rows with the same identifier fields([id, name]). Please make sure the rows are unique.")
+        .isInstanceOf(IllegalStateException.class);
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/TestChangelogIterator.java:202]: can you please add a `.hasMessage()` check here?

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/TestTableSerialization.java [hunk 2]
```diff
@@ -36,29 +39,23 @@
...
-    this.isObjectStoreEnabled = isObjectStoreEnabled;
-  }
+@ExtendWith(ParameterizedTestExtension.class)
+public class TestTableSerialization {
 
-  @Parameterized.Parameters(name = "isObjectStoreEnabled = {0}")
-  public static Object[] parameters() {
-    return new Object[] {"true", "false"};
+  @Parameters(name = "isObjectStoreEnabled = {0}")
+  public static Object[][] parameters() {
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/TestTableSerialization.java:51]: probably better to have it less complex:
```
public static List<String> parameters() {
    return Arrays.asList("true", "false");
  }
```

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/TestTableSerialization.java [hunk 3]
```diff
@@ -36,29 +39,23 @@
...
+    return new Object[][] {{"true"}, {"false"}};
   }
 
   private static final HadoopTables TABLES = new HadoopTables();
 
-  private final String isObjectStoreEnabled;
+  @Parameter(index = 0)
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/TestTableSerialization.java:57]:
```suggestion
  @Parameter
```

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/TestFunctionCatalog.java [hunk 4]
```diff
@@ -27,107 +30,106 @@
...
-  @After
+  @AfterEach
   public void dropDefaultNamespace() {
     sql("DROP NAMESPACE IF EXISTS %s", catalogName + ".default");
   }
 
-  @Test
+  @TestTemplate
   public void testListFunctionsViaCatalog() throws NoSuchNamespaceException {
-    Assertions.assertThat(asFunctionCatalog.listFunctions(EMPTY_NAMESPACE))
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/TestFunctionCatalog.java:59]: these changes are adding lots of noise. Using static imports in new test files is preferrable, but we typically don't want to add too much changes to new PRs by adding static imports. It's ok to keep these now, but just a reminder for any future PRs

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/TestFunctionCatalog.java [hunk 5]
```diff
@@ -27,107 +30,106 @@
...
-  public TestFunctionCatalog() {
+  @BeforeEach
+  public void before() {
+    super.before();
     this.asFunctionCatalog = castToFunctionCatalog(catalogName);
+    createDefaultNamespace();
   }
 
-  @Before
   public void createDefaultNamespace() {
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/TestFunctionCatalog.java:51]: no need to have this separate method now. Can you just move `sql("CREATE NAMESPACE IF NOT EXISTS %s", catalogName + ".default");` to `before()`?

**Comment by nastra** [comment]
@chinmay-bhat this is almost ready to go, just a few minor things to address

**Comment by chinmay-bhat** [comment]
Thanks for the review, I've updated the PR with the suggestions.

**Review by nastra — APPROVED** [review]

**Pull request merged**